### ISSUE TITLE
[BUG]: accessibility-checker checks contrast differently compared to the extension counterpart

### ISSUE THREAD

**Issue by josippapez** [IBMa/equal-access#2320]
### Project

accessibility-checker for Node

### Browser

Chrome

### Operating system

MacOS

### Description

I've encountered a strange issue. I've set up a accessibility-checker packages to work with puppeteer and I get this discrepancy between the results when compared to extension. It seems that the package doesn't check the css and inline styles propperly and always prefers the inline styles approach instead.

### Steps to reproduce

For example:

1. set a color for some html element and set it to !important. Make sure it passes the contrast check.
2. Set the inline style that sets color class to something else that doesn't pass contast check.
```css
.carshowcase__content-type {
    color: black !important; // passes contrast check
}
```
```html
<div
  class="carshowcase__content-type"
  style="color: white"></div> // doesn't pass contrast check but it gets recognized
// as the current color instead of the one in class
```

Expected:
The package checks propperly if the color set inside class has higher importance than the inline and checks against it instead of always defaulting to the inline styles one.

My current workaround (since I think it skips the variables completely):
```scss
.carshowcase__content-type {
    --color: #14612f; // the default one which will be overridden with the inline styles color

    color: var(--color) !important;
}
```
```html
<div
  class="carshowcase__content-type"
  style="--color: {{
     model.typeColor
  }}"></div>
```

**Comment by shunguoy** [comment]
@josippapez  Please note this: (1) Inline styles generally have a higher priority than the same styles from a stylesheet. This means they will override rules defined in external or internal stylesheets for the same properties. (2) Embedded browsers (e.g., Puppeteer) work differently from their browser counterparts (e.g., Chrome) for certain styles in some cases, related to the versions, too, although I haven't tested this specifically for color style.

**Comment by josippapez** [comment]
> @josippapez  Please note this: (1) Inline styles generally have a higher priority than the same styles from a stylesheet. This means they will override rules defined in external or internal stylesheets for the same properties. (2) Embedded browsers (e.g., Puppeteer) work differently from their browser counterparts (e.g., Chrome) for certain styles in some cases, related to the versions, too, although I haven't tested this specifically for color style. 

Yes, i am aware of the differences and how it works but this is something that creates a big difference between test with extension and automation which basically makes it far less useful. Also, isn't latest version of puppeteer using "real" chrome/firefox?

**Comment by MHoov** [comment]
@tombrunet can you respond? I believe in the past you have mentioned it is not uncommon to see different results between the extension and packages.

**Comment by shunguoy** [comment]
@josippapez  An automated tool with an embedded browser helps you identify most accessibility errors; however, you have to use a real browser to do your final test for any other errors an end user might have.

**Comment by tombrunet** [comment]
@josippapez The automated tool and the browser extension inject the same javascript code into Chrome for execution. Given equivalent content (DOM & CSS), they will provide equivalent results. It's identical code, so given identical environments, you'll get identical results.

Scenarios that we've encountered that give different results have always been due to differences in the environment setup and are unrelated to the checker itself. Some examples:
* Using different browsers that process CSS differently (e.g., Chrome vs Firefox vs JSDOM - Chrome and Firefox are typically the same these days, but there are some corner cases).
* Using different versions of the engine (i.e. archives)
* Using different versions of the browser
* Different OS's (usually due to different default color settings)
* Using different widths of the browser that cause different CSS classes to apply

There are other examples, but given that the environment matches, there's no difference between the extension and Puppeteer checking.

**Comment by josippapez** [comment]
@tombrunet thanks for the explanation. I've encountered this particular issue only when testing with the node package. I wanted to make sure I didn't get anything wrong so I checked with the extension in Chrome and Firefox on the same page and got the same result, but only the node package gave different result (mostly because od these contrast issues and even after pointing to the chrome executable that is locally installed on my machine and usting on all of these latest archive version). I'll try to make a git repo that reproduces this particular just in case

**Comment by josippapez** [comment]
@tombrunet  I've managed to reproduce it even though for some reason they are not treated as violations when put in isolation?!

Currently testing on Mac. It seems that the !important is not working well with node package one.

repo: https://github.com/josippapez/test-accessibility

The node package check:

<img width="915" alt="Image" src="https://github.com/user-attachments/assets/ee9440a6-43ef-48a9-9ca3-5f17643f2a03" />

The extension check (same both on firefox and chrome):

![Image](https://github.com/user-attachments/assets/dfee2ce3-4c07-4f25-972c-c7e73e979e28)

**Comment by tombrunet** [comment]
The reason this is happening is that you're copying the content out and then pushing that into the checker. The checker is loading the content into its own context, which cannot resolve the css file because the relative URL doesn't resolve outside of the server context.

You should just pass the Page into the checker, like in this boilerplate example:
https://github.com/IBMa/equal-access/blob/main-4.x/accessibility-checker/boilerplates/mocha-puppeteer-ts/test-ts/basic.test.ts#L37

**Comment by josippapez** [comment]
@tombrunet Thanks for the reply, that worked like a charm! I'll close this issue then with this.

For future reference. tldr;
The problem was in my case i extracted the html with `await page.content();` instead of passing the whole `page` object to the `getCompliance` which resulted in correct results
